src/main/python/run_regression.py

```diff
@@ -0,0 +1,142 @@
+"""
+Anserini: An information retrieval toolkit built on Lucene
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from __future__ import print_function
+import os
+import yaml
+import itertools
+from subprocess import call, check_output
+import argparse
+
+OKBLUE = '\033[94m'
+FAIL = '\033[91m'
+ENDC = '\033[0m'
+
+def construct_indexing_command(yaml_data):
+  """Construct the Anserini indexing command for regression test
+  Args:
+    yaml_data (dict): The yaml config read from config file.
+
+  Returns:
+    (:obj:`list` of :obj:`str`): The command as a list that can be run by calling subprocess.call(command)
+  """
+  index_command = [
+    os.path.join(yaml_data['root'], yaml_data['index_command']),
+    '-collection', yaml_data['collection'],
+    '-generator', yaml_data['generator'],
+    '-threads', str(yaml_data['threads']),
+    '-input', yaml_data['input'],
+    '-index', yaml_data['name']+'_index',
+    *(['-'+c for c in yaml_data['index_options']])
+  ]
+  return index_command
+
+def construct_ranking_command(yaml_data, build_index=True):
+  """Construct the Anserini ranking commands for regression test
+  Args:
+    yaml_data (dict): The yaml config read from config file.
+    build_index (bool): If the index is not built by this script then read the index path for config
+
+  Returns:
+    (:obj:`list` of :obj:`list` of :obj:`str`):
+    The ranking commands as several commands that can be run by calling subprocess.call(command)
+  """
+  ranking_commands = [
+    [
+      os.path.join(yaml_data['root'], yaml_data['search_command']),
+      '-topicreader', yaml_data['topic_reader'],
+      '-index', os.path.join(yaml_data['index_root'] if yaml_data['index_root'] else "", yaml_data['name']+'_index')
```

Comment:
nit: Keep quotes consistent throughout, if single quotes `''` are used for strings use it everywhere.

Reply:
sure